nipype/pipeline/engine/test_newengine/test_single_node.py

```diff
@@ -0,0 +1,123 @@
+def test_singlenode_1(inputs_dict, expected_output):
+    N  = Node(inputs=inuts_dict, mapper="a", interface=fun1)
+    N.run()
+    
+    for (i, out) in enumerate(N.output["out"]):
+        assert out[0] == expected_output[i][0] # checking state values
+        assert (out[1] == expected_output[i][1]).all() # assuming that output value is an array (all() is used)
+
+    
+
+@pytest.mark.parametrize("inputs_dict, expected_output", [
+        ({"a": [3, 4, 5]}, [(["a=3"], 9), (["a=4"], 16), (["a=5"], 25)]),
+        ({"a": np.array([3, 4, 5])}, [(["a=3"], 9), (["a=4"], 16), (["a=5"], 25)]),
+        ])
+def test_singlenode_1a(inputs_dict, expected_output):
+    N  = Node(mapper="a", interface=fun1)
+    # testing if you can set inputs outside __init__
+    N.inputs = inputs_dict
+    N.run()
+
+    for (i, out) in enumerate(N.output["out"]):
+        assert out[0] == expected_output[i][0] 
+        assert (out[1] == expected_output[i][1]).all()
+
+
+
+@pytest.mark.xfail
+def test_single_node_1b():
+    # how to name or rename (from a default name, e.g. "out") outputs 
+    sn = Node(interface=my_function_1, mapper='a', output_name=["out_1"])
+    sn.inputs = {"a" : [3, 1, 8]}
+    sn.run()
+    assert (sn.outputs["out_1"] == [(["a=3"], 9), (["a=1"], 1), (["a=8"], 64)]).all()
+
+
+@pytest.mark.parametrize("inputs_dict, expected_output", [
+        ({"a": [3, 4, 5]}, [(["a=3"], [1, 3, 9, 27]), (["a=4"], [1, 4, 16, 64]), (["a=5"], [1, 5, 25, 125])]),
+        ])
+def test_singlenode_2(inputs_dict, expected_output):
+    N  = Node(inputs=inputs_dict, mapper="a", interface=fun2)
+    N.run()
+
+    for (i, out) in enumerate(N.output["out"]):
+        assert out[0] == expected_output[i][0]
+        assert (out[1] == expected_output[i][1]).all()
+
+
+
+@pytest.mark.parametrize("inputs_dict, expected_output", [
+        ({"a":[3, 1, 8], "b":[0, 1, 2]}, [(["a=3","b=0"], 0), (["a=1","b=1"], 1), (["a=8","b=2"], 16)]),
+        ({"a":[3, 1, 8], "b":[2]}, [(["a=3","b=2"], 6), (["a=1","b=2"], 2), (["a=8","b=2"], 16)]),
+        ])
+def test_single_node_3(inputs_dict, expected_output):
+    N = Node(interface=fun3, mapper=('a','b'))
+    N.inputs = inputs_dict
+    N.run()
+
+    for (i, out) in enumerate(N.output["out"]):
+        assert out[0] == expected_output[i][0]
+        assert (out[1] == expected_output[i][1]).all()
+
+
+
+@pytest.mark.parametrize("inputs_dict, expected_output", [
+        ({"a":[3, 1], "b":[1, 2, 4]}, [[(["a=3","b=1"], 3), (["a=3","b=2"], 6), (["a=3","b=4"], 12)], 
+                                       [(["a=1","b=1"], 1), (["a=1","b=2"], 2), (["a=1","b=4"], 4)]]),
+        ({"a":[[3, 1], [30, 10]], "b":[1, 2, 4]},
+         [[[(["a=3","b=1"], 3), (["a=3","b=2"], 6), (["a=3","b=4"], 12)], 
+           [(["a=1","b=1"], 1), (["a=1","b=2"], 2), (["a=1","b=4"], 4)]],
+          [[(["a=30","b=1"], 30), (["a=30","b=2"], 60), (["a=30","b=4"], 120)],
+           [(["a=10","b=1"], 10), (["a=10","b=2"], 20), (["a=10","b=4"], 40)]]]),
```

Comment:
I'm not sure about it... The first element `(["a=[3, 1]","b=1"], [3, 1])` looked a bit like mine, but I just broke into pieces, since I assumed that the output should be 3-dim array, that every element is related to one value of `a` and one element of `b`. I also don't know why your second element has a 4-elements list `[3, 1, 3, 1]`.
I will review these examples today, so will try to understand better what you had in mind.